File: exercises/python/starting/cards/animal_deck.py
from animal import Animal
from animal_card import AnimalCard
from random import shuffle


class AnimalDeck:
    def __init__(self):
        self.cards = []
        animals = Animal().animals
        for animal in animals:
            self.cards.append(animal)
            self.cards.append(animal)

    def shuffle(self):
        return shuffle(self.cards)

    def deal(self):
        animal = self.cards.pop(0)
        card = AnimalCard(animal)
        return card

File: exercises/javascript/cards/animal-deck.js
const animals = require('./animal.js').Animal;
const { AnimalCard } = require('./animal-card.js');

class AnimalDeck {
  constructor() {
    this.cards = [];
    for (const animal in animals) {
      this.cards.push(new AnimalCard(animal));
      this.cards.push(new AnimalCard(animal));
    }
  }

  getCards() {
    const result = new Array(this.cards.length);
    for (let i = 0; i < this.cards.length; i++) {
      let card = this.cards[i];
      result[i] = card.toString();
    }
    return result;
  };

  shuffle() {
    for (let i = 0; i < this.cards.length; i++) {
      const indexA = Math.floor(Math.random() * i);
      const indexB = i;
  
      const valueA = this.cards[indexA];
      const valueB = this.cards[indexB];
  
      this.cards[indexA] = valueB;
      this.cards[indexB] = valueA;
    }
  };

  deal() {
    const card = this.cards.splice(0, 1)[0];
    return card;
  };
};

module.exports = {
  AnimalDeck
};